### PR TITLE
Rails Capstone project: Maintenance Budget App

### PR DESCRIPTION
**Pull request by mckent05**
## In this PR, I have done the following:
Interactions

Sign up and login pages
The user should be able to register in the app with full name, email, and password (all mandatory).
The user can log into the app using email and password.
If the user is not logged in, they can't access pages that require the user to be logged in (all the pages described below).

Home page (categories page)

When the user logs in, they are presented with the categories page.

For each category, the user can see their name, icon, and the total amount of all the transactions that belong to that category.

When the user clicks (or taps) on a category item, the application navigates to the transactions page for that category.

There is a button "add a new category" at the bottom that brings the user to the page to create a new category.

Transactions page

For a given category, the list of transactions is presented, ordered by the most recent.
At the top of the page, the user could see the total amount for the category (sum of all of the amounts of the transactions in that category).
There is a button "add a new transaction" at the bottom that brings the user to the page to create a new transaction.
When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new category" page
The user fills out a form to create a new category, indicating their name and icon (both mandatory).
The user clicks (or taps) the "Add Category" button to create the new category, and is taken to the home page on success.
When the user clicks on the "Back" button (<), the user navigates to the home page.

"Add a new transaction" page
The user fills out a form to create a new transaction with:
name (mandatory)
amount (mandatory)
categories (mandatory at least one)
The user clicks (or taps) the "Add Transaction" button to create the new category, and is taken to the transactions page for that category.
When the user clicks on the "Back" button (<--), the user navigates to the transactions page for that category.

Testing requirements

Create unit and integration tests for all the most important components of my RoR application.

Technical requirements

I have used Postgres as my database.
Use devise for authentication.
Validate all user input to make sure that anyone with bad intentions cannot compromise my app.
The project is deployed and accessible online.

### CODE REVIEW

File: app/views/equipments/new.html.erb [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+      <div class="field3 d-flex a-center j-center f-col">
+        <%= f.label :name, 'Name'  %>
+        <%= f.text_field :name, :required=>true %>
+      </div>
+    <div class="action1 d-flex a-center f-col">
+         <%= f.submit "Add Equipment"%>
+    </div>
+      
+<% end %>
+</div>
```
Comment (Emmyn5600) [inline app/views/equipments/new.html.erb]: - [ ] When the user clicks on the "Back" button (<), the user navigates to the home page, please implement the back button when it comes to the mobile version.

image

File: app/views/equipments/show.html.erb [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+           
+            <%if can? :destroy, eq.repair %>
+              <%= button_to "Delete", repair_path(eq.repair.id), :method => :delete %>
+            <%end%>
+        </div>
+    <%end%>
+    </div>
+    
+   </div>
+<%end%>
```
Comment (Emmyn5600) [inline app/views/equipments/show.html.erb]: - [ ] When I click on the add a repair, I am getting this error that is coming from the `repairs_controller.rb:6`, please fix it.
 
image

File: app/views/repairs/new.html.erb [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+       <div class="field3 d-flex a-center j-center f-col">
+        <%= f.label :amount, 'Amount'  %>
+        <%= f.number_field :amount, :required=>true %>
+      </div>
+    <div class="action1 d-flex a-center f-col">
+         <%= f.submit "Add Repair"%>
+    </div>
+      
+<% end %>
+</div>
```
Comment (Emmyn5600) [inline app/views/repairs/new.html.erb]: - [ ]  When the user clicks on the "Back" button (<), the user navigates to the home page, please implement the back button when it comes to the mobile version.

image

**Review by Emmyn5600 — CHANGES_REQUESTED** [review]
Hi @mckent05 

Good job so far! I know you have worked hard on making this milestone But  There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight :champagne:

- All linters are passing :heavy_check_mark:
- Good commit history :heavy_check_mark:
- Good PR description :heavy_check_mark:
- The database is well structured/created. :heavy_check_mark:
- You have implemented almost all the requirements. :heavy_check_mark:
- All the tests are passing successfully:heavy_check_mark:
image

## Required Changes :recycle:

- [ ] Check the comment under the review

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!:clap::clap::clap:

#### Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
#### **_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**
------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app/views/equipments/show.html.erb (hunk 2 above)
Comment (DeliceLydia) [reply app/views/equipments/show.html.erb]: Addressed this

File: app/views/repairs/new.html.erb (hunk 3 above)
Comment (DeliceLydia) [reply app/views/repairs/new.html.erb]: Well done!

**Review by DeliceLydia — APPROVED** [review]
## Hi @mckent05, 

In my opinion, your project is ready for the presentation!  There is nothing else to say other than... it's time to merge it :shipit: 
# Congratulations🎉🎉 

### Good luck! 🎉 


Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**